File: app/src/main/java/github/umer0586/sensorserver/util/WifiUtil.java
package github.umer0586.sensorserver.util;

import android.content.Context;
import android.net.wifi.WifiManager;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

public class WifiUtil {

    public static boolean isHotspotEnabled(Context context)
    {
        WifiManager wifi = (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        Method[] wmMethods = wifi.getClass().getDeclaredMethods();

        //is wifi access point (hotspot) enabled
        boolean isWifiAPenabled = false;

        for (Method method : wmMethods)
        {
            if (method.getName().equals("isWifiApEnabled"))
            {

                try
                {
                    method.setAccessible(true); //in the case of visibility change in future APIs
                    isWifiAPenabled = (boolean) method.invoke(wifi);

                } catch (IllegalArgumentException | IllegalAccessException | InvocationTargetException e)
                {
                    e.printStackTrace();
                }
            }
        }
        return isWifiAPenabled;
    }

    public static boolean isWifiEnabled(Context context)
    {
        WifiManager wifiManager = (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        return wifiManager.isWifiEnabled();
    }



}

File: app/src/main/java/github/umer0586/sensorserver/util/WifiUtil.kt
package github.umer0586.sensorserver.util

import android.content.Context
import android.net.wifi.WifiManager
import java.lang.reflect.InvocationTargetException
import java.lang.reflect.Method

object WifiUtil
{


    fun isHotspotEnabled(context: Context?): Boolean
    {
        val wifi =
            context!!.applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
        val wmMethods: Array<Method> = wifi.javaClass.getDeclaredMethods()

        //is wifi access point (hotspot) enabled
        var isWifiAPenabled = false
        for (method in wmMethods)
        {
            if (method.name == "isWifiApEnabled")
            {
                try
                {
                    method.isAccessible = true //in the case of visibility change in future APIs
                    isWifiAPenabled = method.invoke(wifi) as Boolean
                }
                catch (e: IllegalArgumentException)
                {
                    e.printStackTrace()
                }
                catch (e: IllegalAccessException)
                {
                    e.printStackTrace()
                }
                catch (e: InvocationTargetException)
                {
                    e.printStackTrace()
                }
            }
        }
        return isWifiAPenabled
    }

    fun isWifiEnabled(context: Context?): Boolean
    {
        val wifiManager =
            context!!.applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
        return wifiManager.isWifiEnabled
    }
}